src/crimson/osd/ops_executer.cc

```diff
@@ -557,6 +557,10 @@ OpsExecuter::interruptible_errorated_future<OpsExecuter::osd_op_errorator>
 OpsExecuter::execute_op(OSDOp& osd_op)
 {
   head_os = obc->obs;
+  if (pg->get_pgpool().info.has_flag(
+      pg_pool_t::FLAG_WRITE_FADVISE_DONTNEED)) {
+    osd_op.op.flags = osd_op.op.flags | CEPH_OSD_OP_FLAG_FADVISE_DONTNEED;
```

Comment:
Looks to me like this should only affect WRITE and WRITEFULL.  How did the test actually fail, btw?

Reply:
We hit abort with  `stderr:Error: finished tid 1 when last_acked_tid was 4` (Failing tid is changing in different runs).
I updated the PR description since this change doesn't actually provide a fix (yet).